ChustaSoft.Tools.Authorization/Configuration/ConfigurationHelper.cs

```diff
@@ -68,6 +100,11 @@ public static IdentityBuilder WithCustomUserAction<TCustomUserAction>(this Ident
 
 
         #region Private methods
+        private static bool SettingsHaveValues((ExternalAuthenticationProviders Provider, (string ClientId, string ClientSecret)) externalAuthenticationSetting)
+        {
+            return !string.IsNullOrEmpty(externalAuthenticationSetting.Item2.ClientId)
```

Comment:
If tuples params has custom names you can override Item1, Item2, ... 
Semantic sugar :)